core/models.py

```diff
@@ -0,0 +1,99 @@
+from django.db import models
+# Create your models here.
+
+HACKATHON_STATUS_CHOICES = (
+    ('U', 'upcoming'),
+    ('F', 'finished'),
+    ('C', 'cancelled'),
+    ('O', 'ongoing')
+)
+
+class UserProfile(models.Model):
+    """
+      Dummy profile model actual model need to be imported from authentication app.
+      Created because currently authentication pr is not merged. 
+    """
+    pass
+
+
+class Hackathon(models.Model):
+    """
+      A model representing hackathon.
+      `
+        title = models.CharField(default="No title provided", max_length=100)
+        date = models.DateField(auto_now=False, auto_now_add=False)
+        time = models.TimeField(auto_now=False, auto_now_add=False)
+        image = models.CharField(null=True, blank=True, max_length=200)
+        organisers = models.ForeignKey(UserProfile, on_delete=models.SET_NULL)
+        result = 1# to be decided.
+        status = models.CharField(choices=HACKATHON_STATUS_CHOICES, max_length=1)
+      `
+    """
+    title = models.CharField(default="No title provided",
+                             max_length=100, unique=True)
+    date = models.DateField(auto_now_add=False)
+    time = models.TimeField(auto_now_add=False)
+    image = models.CharField(null=True, blank=True, max_length=200)
+    organizers = models.ForeignKey(UserProfile, null=True, on_delete=models.SET_NULL, related_name="hackathon_organizer")
```

Comment:
For now, I think we can allow only the admins to alter the results etc. So we can remove this

Reply:
@nishantwrp
We are having endpoints for craeting and updating the hackathon so I think we can keep it if we write appropiate permission for the endpoint...

Reply:
Yup, that's what I'm saying for simplicity for now we can allow only superuser to update and create the hackathon